draft-ietf-tls-esni.md

```diff
@@ -663,10 +663,12 @@ structure (see {{RFC8446, Section 4.2.11}}) as follows. For each PSK identity
 advertised in the ClientHelloInner, the client generates a random PSK identity
 with the same length. It also generates a random, 32-bit, unsigned integer to
 use as the `obfuscated_ticket_age`. Likewise, for each inner PSK binder, the
-client generates random string of the same length.
+client generates a random string of the same length.
 
-If the server replies with a "pre_shared_key" extension in its ServerHello,
-then the client MUST abort the handshake with an "illegal_parameter" alert.
+Per the rules of {{real-ech}}, the client is not permitted to resume a
```

Comment:
```suggestion
Per the rules of {{real-ech}}, the server is not permitted to resume a
```

It's true that the _client_ can't send a real ticket in the outer CH if it's attempting ECH, but it's the _server_ that actually determines whether or not to resume.